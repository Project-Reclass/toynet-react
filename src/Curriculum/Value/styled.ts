/*
Copyright (C) 1992-2021 Free Software Foundation, Inc.

This file is part of ToyNet React.

ToyNet React is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License as published by the Free
Software Foundation; either version 3, or (at your option) any later
version.

ToyNet React is distributed in the hope that it will be useful, but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
for more details.

You should have received a copy of the GNU General Public License
along with ToyNet React; see the file LICENSE.  If not see
<http://www.gnu.org/licenses/>.  */
import styled from '@emotion/styled';

const StyledReflection = styled.h1`
  font-size: 25px;
  color: white;
  font-weight: bold;
  margin: 1.25rem auto;
`;

const StyledBox = styled.div`
  width: auto;
  border: 2px solid white;
  padding: 20px;
  margin: 1.25rem auto;
  font-style: italic;

  h2 {
    text-align: center;
    color: white;
    font-weight: bold;
  }

  p {
    color: white;
    text-align: center;
  }
`;

const StyledTextArea = styled.textarea`
  background-color: #bbd3ea;
  width: 100%;
  height: 150px;
  border: 2px solid white;
  padding: 10px;
  margin: 1.25rem auto;


  ::placeholder {
    color: black;
  }
`;

const StyledSavebutton = styled.button`
  background-color: #6ca2d8;
  border: 2px solid white;
  color: white;
  padding: 10px 32px;
  text-align: center;
  font-size: 16px;
  margin: 1.25rem auto;
  border-radius: 10px;

  float: right;
`;

export {
  StyledReflection,
  StyledBox,
  StyledTextArea,
  StyledSavebutton,
};
