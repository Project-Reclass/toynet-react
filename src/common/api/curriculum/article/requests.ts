/*
Copyright (C) 1992-2021 Free Software Foundation, Inc.

This file is part of ToyNet React.

ToyNet React is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License as published by the Free
Software Foundation; either version 3, or (at your option) any later
version.

ToyNet React is distributed in the hope that it will be useful, but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
for more details.

You should have received a copy of the GNU General Public License
along with ToyNet React; see the file LICENSE.  If not see
<http://www.gnu.org/licenses/>.

*/
import axios from 'axios';

export interface ArticleImage {
  file: string;
  credit: string;

  caption?: string;
}

export interface ArticleMeta {
  source: string;
  title: string;
  author: string;
  images: ArticleImage[];
}

export const getArticleText = async (articleId: number): Promise<string> => {
  const { data } = await axios.get(`/data/article/${articleId}/article.md`);
  return data;
};

export const getArticleMeta = async (articleId: number): Promise<ArticleMeta> => {
  const { data } = await axios.get(`/data/article/${articleId}/meta.json`);
  return data;
};