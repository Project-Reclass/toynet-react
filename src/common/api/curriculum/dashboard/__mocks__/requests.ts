/*
Copyright (C) 1992-2021 Free Software Foundation, Inc.

This file is part of ToyNet React.

ToyNet React is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License as published by the Free
Software Foundation; either version 3, or (at your option) any later
version.

ToyNet React is distributed in the hope that it will be useful, but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
for more details.

You should have received a copy of the GNU General Public License
along with ToyNet React; see the file LICENSE.  If not see
<http://www.gnu.org/licenses/>.

*/

import { DashboardIntf } from 'src/common/types/curriculum';

const defaultMockData = {
  id: 1,
  name: 'test',
  introduction: 'intro',
  modules: [],
};

export let mockData: DashboardIntf = {...defaultMockData};

export function __resetMock() {
  mockData = {...defaultMockData};
}

export function __setMockData(data: DashboardIntf) {
  mockData.id = data.id;
  mockData.name = data.name;
  mockData.introduction = data.introduction;
  mockData.modules = data.modules;
}

export const getCurriculum = async (): Promise<DashboardIntf | undefined> => {
  return Promise.resolve(mockData);
};