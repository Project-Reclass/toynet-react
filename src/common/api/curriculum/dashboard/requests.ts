import axios from 'axios';
import { DashboardIntf } from 'src/common/types/curriculum';

export const getCurriculum = async (curriculumId: number): Promise<DashboardIntf | undefined> => {
  const { data } = await axios.get(`/data/curriculum/${curriculumId}.json`);
  return data;
};