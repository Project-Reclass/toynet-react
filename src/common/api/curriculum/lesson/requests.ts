import axios from 'axios';

type LessonSlideURI = string;
export interface Meta {
  numSlides: number;
}

export const getLessonSlide = async (submoduleId: number, slideId: number): Promise<LessonSlideURI> => {
  const { data } = await axios.get(`/lesson/${submoduleId}/${slideId}.png`);
  return data;
};

export const getLessonMeta = async (submoduleId: number): Promise<Meta | undefined> => {
  const { data } = await axios.get(`/lesson/${submoduleId}/meta.json`);
  return data;
};
