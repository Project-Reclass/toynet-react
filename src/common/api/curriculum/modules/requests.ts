import { simulateNetwork } from 'src/common/utils';
import modules from 'src/__data__/curriculum/modules.json';

export const getAllModules = () => {
  return simulateNetwork(() => modules);
};

export const getModuleById = (id: number) => {
  return simulateNetwork(() => modules.find(module => module.id === id));
};

export const getModuleByName = (name: string) => {
  return simulateNetwork(() => modules.find(module => module.name === name));
};