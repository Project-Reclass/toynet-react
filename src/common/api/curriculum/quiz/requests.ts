import axios from 'axios';

export const getQuizMeta = async (quizId: number) => {
	const { data } = await axios.get(`/data/quiz/${quizId}/meta.json`);
	return data;
};