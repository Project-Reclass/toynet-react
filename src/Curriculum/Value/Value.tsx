/*
Copyright (C) 1992-2021 Free Software Foundation, Inc.

This file is part of ToyNet React.

ToyNet React is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License as published by the Free
Software Foundation; either version 3, or (at your option) any later
version.

ToyNet React is distributed in the hope that it will be useful, but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
for more details.

You should have received a copy of the GNU General Public License
along with ToyNet React; see the file LICENSE.  If not see
<http://www.gnu.org/licenses/>.

*/
import React from 'react';
import { useParams } from 'react-router-dom';

import { useValueMeta } from '../../common/api/curriculum/value';
import { StyledReflection, StyledBox, StyledTextArea, StyledSavebutton } from './styled';

interface Params {
  valueId: string;
};

const Value = () => {
  const { valueId } = useParams<Params>();

  const { data } = useValueMeta(Number(valueId));

  const placeholderText = `What does ${data?.value.toLowerCase()} mean to you?`;

  return (
    <div className='container-1920 mx-auto' style={{ padding: '1rem' }}>
      <StyledReflection>Reflection: {data?.value}</StyledReflection>

      {data?.inspiration.map((org) => (
        <StyledBox key={org.organization} data-testid="styled-box">
          <h2>{org.organization}</h2>
          <p>{org.definition}</p>
        </StyledBox>
      ))}

      <StyledTextArea placeholder={placeholderText}></StyledTextArea>

      <StyledSavebutton>Save</StyledSavebutton>
    </div>
  );
};

export default Value;
