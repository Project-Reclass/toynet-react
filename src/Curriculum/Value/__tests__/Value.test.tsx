/*
Copyright (C) 1992-2021 Free Software Foundation, Inc.

This file is part of ToyNet React.

ToyNet React is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License as published by the Free
Software Foundation; either version 3, or (at your option) any later
version.

ToyNet React is distributed in the hope that it will be useful, but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
for more details.

You should have received a copy of the GNU General Public License
along with ToyNet React; see the file LICENSE.  If not see
<http://www.gnu.org/licenses/>.

*/
import React from 'react';
import { render, waitFor } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';

import Value from 'src/Curriculum/Value';
import { getValueMeta } from 'src/common/api/curriculum/value/requests';
import { renderWithTheme } from 'src/common/test-utils/renderWithTheme';

jest.mock('src/common/api/curriculum/value/requests.ts');
const getValueMetaMock = getValueMeta as jest.MockedFunction<typeof getValueMeta>;

const RenderWithRouter = ({ children, valueId }: { children: React.ReactChild, valueId: number }) => (
  <MemoryRouter initialEntries={[`/value/${valueId}`]}>
    <Route path="/value/:valueId">{children}</Route>
  </MemoryRouter>
);

describe('The value page', () => {
  it('should render multiple inspirations for integrity', async () => {
    const data = {
        'value': 'Integrity',
        'inspiration': [
            {
                'organization': 'U.S. Air Force',
                'definition': 'Integrity is the adherence to a strong moral code and consistency in one’s actions and values [...] Airmen should be guided by a deeply held sense of honor, not one of personal comfort or uncontrolled selfish appetites.',
            },
            {
                'organization': 'U.S. Army',
                'definition': 'Do what’s right, legally and morally. Integrity is a quality you develop by adhering to moral principles. It requires that you do and say nothing that deceives others. As your integrity grows, so does the trust others place in you […] and, finally, the fundamental acceptance of yourself.',
            },
            {
                'organization': 'U.S. Coast Guard',
                'definition': 'Integrity is our standard. We demonstrate uncompromising ethical conduct and moral behavior in all of our personal actions. We are loyal and accountable to the public trust.',
            },
        ],
    };

    (getValueMeta as jest.MockedFunction<typeof getValueMeta>).mockResolvedValue(data);
    const { getByText, getAllByText, getAllByTestId } = render(
      <RenderWithRouter valueId={5001}>
        <Value />
      </RenderWithRouter>,
    );
    await waitFor(() => getAllByText(/integrity/i));

    expect(getAllByText(/U.S. Air Force/i).length).toBeGreaterThan(0);
    expect(getAllByText(/U.S. Army/i).length).toBeGreaterThan(0);
    expect(getAllByText(/U.S. Coast Guard/i).length).toBeGreaterThan(0);

    expect(getByText(/Integrity is the adherence/i)).toBeInTheDocument();
    expect(getByText(/Do what’s right, legally/i)).toBeInTheDocument();
    expect(getByText(/Integrity is our standard./i)).toBeInTheDocument();

    expect(getAllByTestId('styled-box').length).toEqual(data.inspiration.length);
  });

  it('should render multiple inspirations for respect', async () => {
    const data = {
        'value': 'Respect',
        'inspiration': [
            {
                'organization': 'Army',
                'definition': 'Treat people as they should be treated [...] Respect is what allows us to appreciate the best in other people. Respect is trusting that all people have done their jobs and fulfilled their duty.',
            },
        ],
    };

    getValueMetaMock.mockResolvedValue(data);
    const { getByText, getAllByText, getAllByTestId } = render(
      <RenderWithRouter valueId={5002}>
        <Value />
      </RenderWithRouter>,
    );
    await waitFor(() => getAllByText(/respect/i));

    expect(getAllByText(/Army/i).length).toBeGreaterThan(0);

    expect(getByText(/Treat people as they/i)).toBeInTheDocument();

    expect(getAllByTestId('styled-box').length).toEqual(data.inspiration.length);
  });

  it('should render multiple inspirations for honor', async () => {
    const data = {
        'value': 'Honor',
        'inspiration': [
            {
                'organization': 'Army',
                'definition': 'Live up to Army values [...] Honor is a matter of carrying out, acting, and living the values of respect, duty, loyalty, selfless service, integrity and personal courage in everything you do.',
            },
            {
                'organization': 'Marines / Navy',
                'definition': 'The quality of maturity, dedication, trust, and dependability that commits Marines to act responsibly; to be accountable for actions; to fulfill obligations; and to hold others accountable for their actions.',
            },
        ],
    };
    getValueMetaMock.mockResolvedValue(data);
    const { getByText, getAllByText, getAllByTestId } = render(
      <RenderWithRouter valueId={5003}>
        <Value />
      </RenderWithRouter>,
    );
    await waitFor(() => getAllByText(/honor/i));

    expect(getAllByText(/Army/i).length).toBeGreaterThan(0);
    expect(getAllByText(/Marines/i).length).toBeGreaterThan(0);

    expect(getByText(/Live up to Army values/i)).toBeInTheDocument();
    expect(getByText(/The quality of maturity/i)).toBeInTheDocument();

    expect(getAllByTestId('styled-box').length).toEqual(data.inspiration.length);
  });

  it('should render multiple inspirations for loyalty', async () => {
    const data = {
        'value': 'Loyalty',
        'inspiration': [
            {
                'organization': 'U.S. Army',
                'definition': 'Bear true faith and allegiance to the U.S. Constitution, the Army, your unit and other Soldiers. Bearing true faith and allegiance is a matter of believing in and devoting yourself to something or someone … By wearing the uniform of the U.S. Army you are expressing your loyalty. And by doing your share, you show your loyalty to your unit.',
            },
        ],
    };
    getValueMetaMock.mockResolvedValue(data);
    const { getByText, getAllByText, getAllByTestId } = render(
      <RenderWithRouter valueId={5004}>
        <Value />
      </RenderWithRouter>,
    );
    await waitFor(() => getAllByText(/loyalty/i));

    expect(getAllByText(/U.S. Army/i).length).toBeGreaterThan(0);

    expect(getByText(/Bear true faith and allegiance/i)).toBeInTheDocument();

    expect(getAllByTestId('styled-box').length).toEqual(data.inspiration.length);
  });

  it('should render the same based on URL parameters for integrity', () => {
    const { container } = renderWithTheme(
      <RenderWithRouter valueId={5001}>
        <Value />
      </RenderWithRouter>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render the same based on URL parameters for respect', () => {
    const { container } = renderWithTheme(
      <RenderWithRouter valueId={5002}>
        <Value />
      </RenderWithRouter>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render the same based on URL parameters for honor', () => {
    const { container } = renderWithTheme(
      <RenderWithRouter valueId={5003}>
        <Value />
      </RenderWithRouter>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render the same based on URL parameters for loyalty', () => {
    const { container } = renderWithTheme(
      <RenderWithRouter valueId={5004}>
        <Value />
      </RenderWithRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});