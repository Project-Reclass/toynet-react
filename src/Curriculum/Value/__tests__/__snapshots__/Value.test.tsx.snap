// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The value page should render the same based on URL parameters for honor 1`] = `
<div>
  <div
    class="css-1c3ot3n"
    id="#"
  >
    <div
      class="chakra-container css-jf7n8r"
    >
      <h2
        class="chakra-heading css-1gdkznd"
      >
        Reflection: 
        Honor
      </h2>
      <div
        class="css-1yazsj0"
        data-testid="styled-box"
      >
        <h2>
          Army
        </h2>
        <p>
          Live up to Army values [...] Honor is a matter of carrying out, acting, and living the values of respect, duty, loyalty, selfless service, integrity and personal courage in everything you do.
        </p>
      </div>
      <div
        class="css-1yazsj0"
        data-testid="styled-box"
      >
        <h2>
          Marines / Navy
        </h2>
        <p>
          The quality of maturity, dedication, trust, and dependability that commits Marines to act responsibly; to be accountable for actions; to fulfill obligations; and to hold others accountable for their actions.
        </p>
      </div>
      <textarea
        class="css-21ory"
        placeholder="What does honor mean to you?"
      />
      <button
        class="chakra-button css-f6n2t5"
        type="button"
      >
        Save
      </button>
      <div
        class="chakra-stack css-1w6iwr8"
      >
        <hr
          aria-orientation="horizontal"
          class="chakra-divider css-svjswr"
        />
        <div
          class="css-1lads1q"
        >
          <div
            class="chakra-stack css-jqjin"
          >
            <button
              aria-label="previous"
              class="chakra-button css-1muaiii"
              disabled=""
              type="button"
            >
              <span
                class="chakra-button__icon css-1wh2kri"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-onkibi"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g
                    fill="currentColor"
                  >
                    <path
                      d="M10.416,12a2.643,2.643,0,0,1,.775-1.875L20.732.584a1.768,1.768,0,0,1,2.5,2.5l-8.739,8.739a.25.25,0,0,0,0,.354l8.739,8.739a1.768,1.768,0,0,1-2.5,2.5l-9.541-9.541A2.643,2.643,0,0,1,10.416,12Z"
                    />
                    <path
                      d="M.25,12a2.643,2.643,0,0,1,.775-1.875L10.566.584a1.768,1.768,0,0,1,2.5,2.5L4.327,11.823a.25.25,0,0,0,0,.354l8.739,8.739a1.768,1.768,0,0,1-2.5,2.5L1.025,13.875A2.643,2.643,0,0,1,.25,12Z"
                    />
                  </g>
                </svg>
              </span>
              Previous
            </button>
            <button
              aria-label="next"
              class="chakra-button css-1muaiii"
              disabled=""
              type="button"
            >
              Next
              <span
                class="chakra-button__icon css-1hzyiq5"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-onkibi"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g
                    fill="currentColor"
                  >
                    <path
                      d="M13.584,12a2.643,2.643,0,0,1-.775,1.875L3.268,23.416a1.768,1.768,0,0,1-2.5-2.5l8.739-8.739a.25.25,0,0,0,0-.354L.768,3.084a1.768,1.768,0,0,1,2.5-2.5l9.541,9.541A2.643,2.643,0,0,1,13.584,12Z"
                    />
                    <path
                      d="M23.75,12a2.643,2.643,0,0,1-.775,1.875l-9.541,9.541a1.768,1.768,0,0,1-2.5-2.5l8.739-8.739a.25.25,0,0,0,0-.354L10.934,3.084a1.768,1.768,0,0,1,2.5-2.5l9.541,9.541A2.643,2.643,0,0,1,23.75,12Z"
                    />
                  </g>
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`The value page should render the same based on URL parameters for integrity 1`] = `
<div>
  <div
    class="css-1c3ot3n"
    id="#"
  >
    <div
      class="chakra-container css-jf7n8r"
    >
      <h2
        class="chakra-heading css-1gdkznd"
      >
        Reflection: 
        Integrity
      </h2>
      <div
        class="css-1yazsj0"
        data-testid="styled-box"
      >
        <h2>
          U.S. Air Force
        </h2>
        <p>
          Integrity is the adherence to a strong moral code and consistency in one’s actions and values [...] Airmen should be guided by a deeply held sense of honor, not one of personal comfort or uncontrolled selfish appetites.
        </p>
      </div>
      <div
        class="css-1yazsj0"
        data-testid="styled-box"
      >
        <h2>
          U.S. Army
        </h2>
        <p>
          Do what’s right, legally and morally. Integrity is a quality you develop by adhering to moral principles. It requires that you do and say nothing that deceives others. As your integrity grows, so does the trust others place in you […] and, finally, the fundamental acceptance of yourself.
        </p>
      </div>
      <div
        class="css-1yazsj0"
        data-testid="styled-box"
      >
        <h2>
          U.S. Coast Guard
        </h2>
        <p>
          Integrity is our standard. We demonstrate uncompromising ethical conduct and moral behavior in all of our personal actions. We are loyal and accountable to the public trust.
        </p>
      </div>
      <textarea
        class="css-21ory"
        placeholder="What does integrity mean to you?"
      />
      <button
        class="chakra-button css-f6n2t5"
        type="button"
      >
        Save
      </button>
      <div
        class="chakra-stack css-1w6iwr8"
      >
        <hr
          aria-orientation="horizontal"
          class="chakra-divider css-svjswr"
        />
        <div
          class="css-1lads1q"
        >
          <div
            class="chakra-stack css-jqjin"
          >
            <button
              aria-label="previous"
              class="chakra-button css-1muaiii"
              disabled=""
              type="button"
            >
              <span
                class="chakra-button__icon css-1wh2kri"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-onkibi"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g
                    fill="currentColor"
                  >
                    <path
                      d="M10.416,12a2.643,2.643,0,0,1,.775-1.875L20.732.584a1.768,1.768,0,0,1,2.5,2.5l-8.739,8.739a.25.25,0,0,0,0,.354l8.739,8.739a1.768,1.768,0,0,1-2.5,2.5l-9.541-9.541A2.643,2.643,0,0,1,10.416,12Z"
                    />
                    <path
                      d="M.25,12a2.643,2.643,0,0,1,.775-1.875L10.566.584a1.768,1.768,0,0,1,2.5,2.5L4.327,11.823a.25.25,0,0,0,0,.354l8.739,8.739a1.768,1.768,0,0,1-2.5,2.5L1.025,13.875A2.643,2.643,0,0,1,.25,12Z"
                    />
                  </g>
                </svg>
              </span>
              Previous
            </button>
            <button
              aria-label="next"
              class="chakra-button css-1muaiii"
              disabled=""
              type="button"
            >
              Next
              <span
                class="chakra-button__icon css-1hzyiq5"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-onkibi"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g
                    fill="currentColor"
                  >
                    <path
                      d="M13.584,12a2.643,2.643,0,0,1-.775,1.875L3.268,23.416a1.768,1.768,0,0,1-2.5-2.5l8.739-8.739a.25.25,0,0,0,0-.354L.768,3.084a1.768,1.768,0,0,1,2.5-2.5l9.541,9.541A2.643,2.643,0,0,1,13.584,12Z"
                    />
                    <path
                      d="M23.75,12a2.643,2.643,0,0,1-.775,1.875l-9.541,9.541a1.768,1.768,0,0,1-2.5-2.5l8.739-8.739a.25.25,0,0,0,0-.354L10.934,3.084a1.768,1.768,0,0,1,2.5-2.5l9.541,9.541A2.643,2.643,0,0,1,23.75,12Z"
                    />
                  </g>
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`The value page should render the same based on URL parameters for loyalty 1`] = `
<div>
  <div
    class="css-1c3ot3n"
    id="#"
  >
    <div
      class="chakra-container css-jf7n8r"
    >
      <h2
        class="chakra-heading css-1gdkznd"
      >
        Reflection: 
        Loyalty
      </h2>
      <div
        class="css-1yazsj0"
        data-testid="styled-box"
      >
        <h2>
          U.S. Army
        </h2>
        <p>
          Bear true faith and allegiance to the U.S. Constitution, the Army, your unit and other Soldiers. Bearing true faith and allegiance is a matter of believing in and devoting yourself to something or someone … By wearing the uniform of the U.S. Army you are expressing your loyalty. And by doing your share, you show your loyalty to your unit.
        </p>
      </div>
      <textarea
        class="css-21ory"
        placeholder="What does loyalty mean to you?"
      />
      <button
        class="chakra-button css-f6n2t5"
        type="button"
      >
        Save
      </button>
      <div
        class="chakra-stack css-1w6iwr8"
      >
        <hr
          aria-orientation="horizontal"
          class="chakra-divider css-svjswr"
        />
        <div
          class="css-1lads1q"
        >
          <div
            class="chakra-stack css-jqjin"
          >
            <button
              aria-label="previous"
              class="chakra-button css-1muaiii"
              disabled=""
              type="button"
            >
              <span
                class="chakra-button__icon css-1wh2kri"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-onkibi"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g
                    fill="currentColor"
                  >
                    <path
                      d="M10.416,12a2.643,2.643,0,0,1,.775-1.875L20.732.584a1.768,1.768,0,0,1,2.5,2.5l-8.739,8.739a.25.25,0,0,0,0,.354l8.739,8.739a1.768,1.768,0,0,1-2.5,2.5l-9.541-9.541A2.643,2.643,0,0,1,10.416,12Z"
                    />
                    <path
                      d="M.25,12a2.643,2.643,0,0,1,.775-1.875L10.566.584a1.768,1.768,0,0,1,2.5,2.5L4.327,11.823a.25.25,0,0,0,0,.354l8.739,8.739a1.768,1.768,0,0,1-2.5,2.5L1.025,13.875A2.643,2.643,0,0,1,.25,12Z"
                    />
                  </g>
                </svg>
              </span>
              Previous
            </button>
            <button
              aria-label="next"
              class="chakra-button css-1muaiii"
              disabled=""
              type="button"
            >
              Next
              <span
                class="chakra-button__icon css-1hzyiq5"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-onkibi"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g
                    fill="currentColor"
                  >
                    <path
                      d="M13.584,12a2.643,2.643,0,0,1-.775,1.875L3.268,23.416a1.768,1.768,0,0,1-2.5-2.5l8.739-8.739a.25.25,0,0,0,0-.354L.768,3.084a1.768,1.768,0,0,1,2.5-2.5l9.541,9.541A2.643,2.643,0,0,1,13.584,12Z"
                    />
                    <path
                      d="M23.75,12a2.643,2.643,0,0,1-.775,1.875l-9.541,9.541a1.768,1.768,0,0,1-2.5-2.5l8.739-8.739a.25.25,0,0,0,0-.354L10.934,3.084a1.768,1.768,0,0,1,2.5-2.5l9.541,9.541A2.643,2.643,0,0,1,23.75,12Z"
                    />
                  </g>
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`The value page should render the same based on URL parameters for respect 1`] = `
<div>
  <div
    class="css-1c3ot3n"
    id="#"
  >
    <div
      class="chakra-container css-jf7n8r"
    >
      <h2
        class="chakra-heading css-1gdkznd"
      >
        Reflection: 
        Respect
      </h2>
      <div
        class="css-1yazsj0"
        data-testid="styled-box"
      >
        <h2>
          Army
        </h2>
        <p>
          Treat people as they should be treated [...] Respect is what allows us to appreciate the best in other people. Respect is trusting that all people have done their jobs and fulfilled their duty.
        </p>
      </div>
      <textarea
        class="css-21ory"
        placeholder="What does respect mean to you?"
      />
      <button
        class="chakra-button css-f6n2t5"
        type="button"
      >
        Save
      </button>
      <div
        class="chakra-stack css-1w6iwr8"
      >
        <hr
          aria-orientation="horizontal"
          class="chakra-divider css-svjswr"
        />
        <div
          class="css-1lads1q"
        >
          <div
            class="chakra-stack css-jqjin"
          >
            <button
              aria-label="previous"
              class="chakra-button css-1muaiii"
              disabled=""
              type="button"
            >
              <span
                class="chakra-button__icon css-1wh2kri"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-onkibi"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g
                    fill="currentColor"
                  >
                    <path
                      d="M10.416,12a2.643,2.643,0,0,1,.775-1.875L20.732.584a1.768,1.768,0,0,1,2.5,2.5l-8.739,8.739a.25.25,0,0,0,0,.354l8.739,8.739a1.768,1.768,0,0,1-2.5,2.5l-9.541-9.541A2.643,2.643,0,0,1,10.416,12Z"
                    />
                    <path
                      d="M.25,12a2.643,2.643,0,0,1,.775-1.875L10.566.584a1.768,1.768,0,0,1,2.5,2.5L4.327,11.823a.25.25,0,0,0,0,.354l8.739,8.739a1.768,1.768,0,0,1-2.5,2.5L1.025,13.875A2.643,2.643,0,0,1,.25,12Z"
                    />
                  </g>
                </svg>
              </span>
              Previous
            </button>
            <button
              aria-label="next"
              class="chakra-button css-1muaiii"
              disabled=""
              type="button"
            >
              Next
              <span
                class="chakra-button__icon css-1hzyiq5"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-onkibi"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g
                    fill="currentColor"
                  >
                    <path
                      d="M13.584,12a2.643,2.643,0,0,1-.775,1.875L3.268,23.416a1.768,1.768,0,0,1-2.5-2.5l8.739-8.739a.25.25,0,0,0,0-.354L.768,3.084a1.768,1.768,0,0,1,2.5-2.5l9.541,9.541A2.643,2.643,0,0,1,13.584,12Z"
                    />
                    <path
                      d="M23.75,12a2.643,2.643,0,0,1-.775,1.875l-9.541,9.541a1.768,1.768,0,0,1-2.5-2.5l8.739-8.739a.25.25,0,0,0,0-.354L10.934,3.084a1.768,1.768,0,0,1,2.5-2.5l9.541,9.541A2.643,2.643,0,0,1,23.75,12Z"
                    />
                  </g>
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
